lint/lint.go

```diff
@@ -38,3 +39,24 @@ type Context struct {
 type Checker interface {
 	Check(f *ast.File) []Warning
 }
+
+var checkers = map[string]func(c *Context) Checker{
+	"param-name":        newParamNameChecker,
+	"type-guard":        newTypeGuardChecker,
+	"parenthesis":       newParenthesisChecker,
+	"underef":           newUnderefChecker,
+	"param-duplication": newParamDuplicationChecker,
+}
+
+func NewChecker(name string, ctx *Context) Checker {
```

Comment:
Build failed due to missing comment.
Add something like:
```
NewChecker returns checker that implements check of specified name.
```

Reply:
Done.